### PR TITLE
Rename at file creation

### LINKED ISSUE #10164: File browser -> new file -> rename
## Description

When creating a new file from the context menu, we should allow the user to directly rename the file. It works like that for folders, but not for files.

## Reproduce

filerename

## Expected behavior

The user should be prompt to rename the file directly after the creation

I am happy to open a PR myself.

### PR DESCRIPTION
**Pull request by martinRenou**
## References

Fix #10164

rename

### CODE REVIEW

File: packages/filebrowser/src/browser.ts [hunk 1]
```diff
@@ -273,6 +273,35 @@ export class FileBrowser extends Widget {
...
+    // request.
+    void this._manager
+      .newUntitled({
+        path: this.model.path,
+        type: 'file',
+        ext: options.ext
+      })
+      .then(async model => {
+        await this._listing.selectItemByName(model.name);
+        await this.rename();
```
Comment (martinRenou) [inline packages/filebrowser/src/browser.ts:297]: It is unclear to me why this is needed, that's not needed for directories

**Comment by martinRenou** [comment]
It looks like it was not needed to call `rename()` explicitely for directories because of those lines:
https://github.com/jupyterlab/jupyterlab/blob/f0790b3ffd187b7dc923156b8ae227855abecefb/packages/filebrowser/src/listing.ts#L1562-L1565

so now the question is do you prefer my approach of renaming after creation? Or do I change the "if" condition in the listing flow so that it triggers a renaming for files also?

**Comment by jtpio** [comment]
Thanks Martin for looking into this!

> so now the question is do you prefer my approach of renaming after creation? Or do I change the "if" condition in the listing flow so that it triggers a renaming for files also?

Unless there was a specific reason to have the renaming be done only for directories, maybe it would be better ti have an explicit call to the `rename` method?

**Comment by jtpio** [comment]
Digging a little bit in the history, it looks like the check for `directory` might have been added in https://github.com/jupyterlab/jupyterlab/pull/1273.

Since it's been a while, maybe this is not necessary anymore and both files and directories could be handled in a similar manner?

**Comment by martinRenou** [comment]
I'll update the PR and harmonize this then

**Comment by martinRenou** [comment]
This should be ready

**Review by blink1073 — APPROVED** [review]
Thanks!

**Pull request merged**